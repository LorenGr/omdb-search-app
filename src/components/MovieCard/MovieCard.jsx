import React, { useState } from 'react';
import { Card, Tag } from 'antd';
import defaultImage from '../../assets/default.jpg';

const { Meta } = Card;

const MovieCard = ({ title, type, year, image }) => {

    const [hasError, setError] = useState(false);
    const handleImageError = (e) => {
        setError(true);
    };

    return (
        <Card
            hoverable
            style={{ width: 240 }}
            cover={<img alt={title} src={hasError ? defaultImage : image} onError={handleImageError} />}
        >
            <Meta title={title} />
            <div style={{ marginTop: '10px' }}>
                <Tag color="blue">{type}</Tag>
                <Tag color="green">{year}</Tag>
            </div>
        </Card>
    );
};

export default MovieCard;
