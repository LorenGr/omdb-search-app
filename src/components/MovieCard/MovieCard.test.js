// src/components/__tests__/MovieCard.test.js
import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import MovieCard from './MovieCard';

describe('MovieCard', () => {
    const mockTitle = 'Inception';
    const mockImage = 'https://example.com/inception.jpg';
    const mockType = 'movie';
    const mockYear = '2010';

    it('renders correctly with given props', () => {
        render(<MovieCard title={mockTitle} image={mockImage} type={mockType} year={mockYear} />);

        expect(screen.getByText(mockTitle)).toBeInTheDocument();
        expect(screen.getByAltText(mockTitle)).toHaveAttribute('src', mockImage);
        expect(screen.getByText(mockType)).toBeInTheDocument();
        expect(screen.getByText(mockYear)).toBeInTheDocument();
    });

    // Test for fallback image logic
    it('displays fallback image on image load error', () => {
        render(<MovieCard title={mockTitle} image={mockImage} />);
        const image = screen.getByAltText(mockTitle);
        fireEvent.error(image);
        expect(image).toHaveAttribute('src', 'default.jpg');
    });
});
