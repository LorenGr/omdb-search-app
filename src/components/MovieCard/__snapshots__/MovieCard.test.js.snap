// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCard renders and matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="ant-card ant-card-bordered ant-card-hoverable css-dev-only-do-not-override-dkbvqv"
    style="width: 240px;"
  >
    <div
      class="ant-card-cover"
    >
      <img />
    </div>
    <div
      class="ant-card-body"
    >
      <div
        class="ant-card-meta"
      />
      <div
        style="margin-top: 10px;"
      >
        <span
          class="ant-tag ant-tag-blue css-dev-only-do-not-override-dkbvqv"
        />
        <span
          class="ant-tag ant-tag-green css-dev-only-do-not-override-dkbvqv"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
